src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -615,18 +639,21 @@ protected Task ProduceEnd()
 
             if (!HasResponseStarted)
             {
-                return ProduceEndAwaited();
+                ProduceStart(appCompleted: true);
+                // Force flush
+                var flushTask = SocketOutput.WriteAsync(_emptyData);
```

Comment:
Prevents an unobserved task exception or having to use `#pragma warning disable 4014`